src/main/java/org/broadinstitute/hdf5/HDF5File.java

```diff
@@ -477,21 +510,21 @@ private void checkIsOpen() {
     /**
      * Opens a HDF5 file given its access {@link OpenMode mode}.
      */
-    private static int open(final File file, final OpenMode mode) {
+    private static int open(final Path file, final OpenMode mode) {
         final int fileId;
         try {
             if (mode == OpenMode.CREATE) {
-                file.delete();
-                file.createNewFile();
+                Files.deleteIfExists(file);
+                Files.createFile(file);
             }
-            fileId = H5.H5Fopen(file.getAbsolutePath(), mode.getFlags(), HDF5Constants.H5P_DEFAULT);
+            fileId = H5.H5Fopen(file.toAbsolutePath().toString(), mode.getFlags(), HDF5Constants.H5P_DEFAULT);
```

Comment:
That's exactly what I was thinking, but I hadn't thought through how writing would work.  

I think the second option is probably not feasible.  Identifying when the temp file has been updated and copying it back to the original path would be tricky and poorly performing I think.  

The first suggestion seems reasonable but it's definitely going to be error prone.  People are not always good about closing files.  We'd have to have very clear logging explaining what's going on and aggressively fail if a file fails to be written back correctly. 

What would you think about supporting Paths only for the read only case and throwing an explanatory error if someone tries to use a non-file path with write enabled?